Controller/MediaAdminController.php

```diff
@@ -77,10 +77,12 @@ public function listAction(Request $request = null)
         $datagrid->setValue('context', null, $context);
 
         // retrieve the main category for the tree view
-        $category = $this->container->get('sonata.classification.manager.category')->getRootCategory($context);
+        $rootCategory = $this->container->get('sonata.classification.manager.category')->getRootCategory($context);
+        // This should be safe as the root category has to exist for a given context but I do not like fatal errors
```

Comment:
OK. I just read a little bit more `Sonata\ClassificationBundle\Entity\CategoryManager` : in fact, `CategoryManager::loadCategories()` creates the missing entity if needed ; I missed this part of the code, my mistake.

I generally do not like to create on the fly objects that have to be, but it is the way it is actually done so my "non null" check is not necessary anymore and the exception throwing is not too.